Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1818,11 +1818,11 @@ - (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView
 
 #if ENABLE(ASYNC_SCROLLING)
     // FIXME: We will want to detect whether snapping will occur before beginning to drag. See WebPageProxy::didCommitLayerTree.
-    auto* coordinator = downcast<WebKit::RemoteScrollingCoordinatorProxyIOS>(_page->scrollingCoordinatorProxy());
-    ASSERT(scrollView == _scrollView.get());
-    [_scrollView _setDecelerationRateInternal:(coordinator && coordinator->shouldSetScrollViewDecelerationRateFast()) ? UIScrollViewDecelerationRateFast : UIScrollViewDecelerationRateNormal];
-
-    coordinator->setRootNodeIsInUserScroll(true);
+    if (auto* coordinator = downcast<WebKit::RemoteScrollingCoordinatorProxyIOS>(_page->scrollingCoordinatorProxy())) {
```

Comment:
should this be `dynamicDowncast` instead?

Reply:
It looks like `dynamicDowncast` expects a reference, but the underlying `m_scrollingCoordinatorProxy` member can be `nullptr`

Reply:
> should this be `dynamicDowncast` instead?

It seems that this can only ever be a `RemoteScrollingCoordinatorProxyIOS` since `RemoteScrollingCoordinatorProxy` is pure virtual and the only other subclass, `RemoteScrollingCoordinatorProxyMac`, is not compiled on iOS; this means the existing code is fine as-is.

Reply:
@whsieh do you mean there's no need for a downcast here?

Reply:
What you have above (with just `downcast<>`) should suffice.